core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/V4RequestSigner.java

```diff
@@ -100,11 +106,33 @@ static V4RequestSigner query(V4Properties properties) {
      */
     static V4RequestSigner presigned(V4Properties properties, Duration expirationDuration) {
         return requestBuilder -> {
-            requestBuilder.putRawQueryParameter(SignerConstant.X_AMZ_EXPIRES,
-                                                Long.toString(expirationDuration.getSeconds())
-            );
+            // Add pre-requisites
+            if (properties.getCredentials() instanceof AwsSessionCredentialsIdentity) {
+                requestBuilder.putRawQueryParameter(SignerConstant.X_AMZ_SECURITY_TOKEN,
+                                                    ((AwsSessionCredentialsIdentity) properties.getCredentials()).sessionToken());
+            }
+            // We have to add the host-header here explicitly, since pre-signed request requires it in the signed-header param
+            addHostHeader(requestBuilder);
```

Comment:
Does the v4a signer already do this for presigned case?
 Does v4a signer always include x-amz-content-sha256?

Reply:
CRT would be the thing that manipulates headers - the signer simply delegates to it. However, the signer _does_ configure CRT to always add the hash header, yes: `signingConfig.setSignedBodyHeader(X_AMZ_CONTENT_SHA256)`